packages/sanity/src/desk/structureBuilder/GenericList.ts

```diff
@@ -1,65 +1,86 @@
 import {camelCase} from 'lodash'
-import {InitialValueTemplateItem} from '../../templates'
-import {StructureNode, SerializeOptions, Serializable, Child} from './StructureNodes'
+import {
+  StructureChild,
+  StructureSerializable,
+  StructureSerializeOptions,
+  StructureNode,
+} from './StructureNodes'
 import {layoutOptions} from './Layout'
-import {MenuItem, MenuItemBuilder, maybeSerializeMenuItem} from './MenuItem'
-import {MenuItemGroup, MenuItemGroupBuilder, maybeSerializeMenuItemGroup} from './MenuItemGroup'
-import {IntentChecker, defaultIntentChecker} from './Intent'
+import {StructureMenuItem, StructureMenuItemBuilder, maybeSerializeMenuItem} from './MenuItem'
+import {
+  StructureMenuItemGroup,
+  StructureMenuItemGroupBuilder,
+  maybeSerializeMenuItemGroup,
+} from './MenuItemGroup'
+import {StructureIntentChecker, defaultIntentChecker} from './Intent'
 import {SerializeError} from './SerializeError'
 import {
-  InitialValueTemplateItemBuilder,
+  StructureInitialValueTemplateItemBuilder,
   maybeSerializeInitialValueTemplateItem,
 } from './InitialValueTemplateItem'
 import {validateId} from './util/validateId'
-import {PreviewLayoutKey} from '../../components/previews'
+import {PreviewLayoutKey, InitialValueTemplateItem} from 'sanity'
 
 function noChildResolver() {
   return undefined
 }
 
-export const shallowIntentChecker: IntentChecker = (intentName, params, {pane, index}): boolean => {
+/** @internal */
+export const shallowIntentChecker: StructureIntentChecker = (
```

Comment:
Halfway through these structure renames, I ask myself: Are they necessary? 

These are exported from `sanity/desk` which already explicitly scope to them to that domain. These changes will hit v2 structure code trying to "use the types everywhere" very hard I think. 

Maybe we could just say prepend "Structure" and you'll be fine in release docs, but this looks it could introduce alot of busywork when upgrading to v3.

Reply:
No, I think I regret these changes now.

Reply:
Some of the changes were good, I think:

`Component => StructureComponent` (since it clashes with React).

I guess I thought it made sense to prefix all structure builder members, instead of a select few.

Reply:
Reverting, will let someone else worry about this in the future :-)